samples/Azure.ResourceManager.Sample/Generated/AvailabilitySetResource.Serialization.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.ClientModel.Primitives;
+using System.Text.Json;
+
+namespace Azure.ResourceManager.Sample
+{
+    public partial class AvailabilitySetResource : IJsonModel<AvailabilitySetData>
+    {
+        void IJsonModel<AvailabilitySetData>.Write(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+        {
+            writer.WriteStringValue(ModelReaderWriter.Write(Data, options));
+        }
+
+        AvailabilitySetData IJsonModel<AvailabilitySetData>.Create(ref Utf8JsonReader reader, ModelReaderWriterOptions options)
+        {
+            return ModelReaderWriter.Read<AvailabilitySetData>(new BinaryData(reader.ValueSequence), options);
+        }
+
+        BinaryData IPersistableModel<AvailabilitySetData>.Write(ModelReaderWriterOptions options)
+        {
+            return ModelReaderWriter.Write(Data, options);
+        }
+
+        AvailabilitySetData IPersistableModel<AvailabilitySetData>.Create(BinaryData data, ModelReaderWriterOptions options)
+        {
+            return ModelReaderWriter.Read<AvailabilitySetData>(data, options);
+        }
+
+        string IPersistableModel<AvailabilitySetData>.GetFormatFromOptions(ModelReaderWriterOptions options) => "J";
```

Comment:
this should return Data.GetFormatFromOptions

Reply:
updated